specs/gateway.oag.yaml

```diff
@@ -0,0 +1,20 @@
+document: specs/gateway.yaml
+output: gateway/zz_oag_generated_gateway.go
+
+# Put gateway into a subpackage for now, as it is being introduced, and has
+# some values that conflict with other endpoints
+package:
+  path: github.com/manifoldco/go-manifold/gateway
+  name: gateway
+
+boilerplate:
+  base_url: disabled
+  backend: disabled
+  endpoint: disabled
+  client_prefix: API
```

Comment:
we can get rid of the prefix for now IMO, since its in a subpackage and not sitting next to other APIs

Reply:
Getting rid of the prefix means oag will generate `New()` and `Client` and I need those two names so I can add authentication code to them. I left it as is... alternatively we could use `Gateway` as the prefix like the others but than we'd end up with `gateway.GatewayClient`